### ISSUE TITLE
Incorrectly marking lint within math expressions

### ISSUE THREAD

**Issue by Reilithion** [wooorm/remark-lint#101]
Many markdown syntaxes (syntaxi?) allow inline math expressions enclosed in dollar signs. But it seems remark-lint doesn't treat these environments differently. So while Pandoc will correctly format $f[x]$, remark-lint will complain that it should "Use the trailing [] on reference links" and "Found reference to undefined definition".
I've run into other problems too, but they seem much more difficult to reduce to a simple test case. (Under certain circumstances I haven't pinned down yet, I can get "Emphasis should use * as a marker".)

**Comment by wooorm** [comment]
> Many markdown syntaxes (syntaxi?) allow inline math expressions enclosed in dollar signs.

It’s very non-standard, so you’ll need a separate plugin for that. Something like remark-github, but for math. I tried my hand at that some time ago, but couldn’t find enough input/output examples.
If you can come up with a lot of tests / example implementations, could you open an issue over on remark’s repo? 

> But it seems remark-lint doesn't treat these environments differently.

Handling of math should happen outside of remark-lint.

> So while Pandoc will correctly format $f[x]$, remark-lint will complain that it should "Use the trailing [] on reference links" and "Found reference to undefined definition".

remark-lint checks markdown for commonmark, gfm, or pedantic markdown. IMO it’s right to complain about non-standard stuff.

> I've run into other problems too, but they seem much more difficult to reduce to a simple test case. 

Do let me know!

**Comment by Ir1d** [comment]
@wooorm I wonder if there's a way to ignore all math equations during linting?

I tried this

```
{
  "plugins": [
    "remark-preset-lint-recommended",
    ["remark-lint-no-shortcut-reference-link", false]
  ]
}

```

to avoid the `$f[x]$` being modified.

Then I noticed `$O(\log n)$` is turned into `$O(\\log n)$`. Any idea on how to fix this?

It's not likely possible to add `<!-- ignore -->` around math equations since there  are so many of them

**Comment by wooorm** [comment]
If you’re using [`remark-math`](https://github.com/Rokt33r/remark-math) you should be fine!

**Comment by wooorm** [comment]
If you’re using remark-lint you’re already using remark as well! See the [unified readme](https://github.com/unifiedjs/unified#description) and [website](https://unifiedjs.github.io) for more info on how it all works!

**Comment by Ir1d** [comment]
sorry if i got it wrong.
I mean I couldn't use remark-math to translate math expressions to mathjax text, because I was using python-markdown to render markdown to html.
I simple want to lint the markdown files before rendering, now remark-lint converts `\` to `\\`, I understand that it works fine when rendering with remark-math, but I hope this convert could be disabled so that the math expression would still work fine with other parser.
As far as I know, `\\` is regarded as a linebreak in many parsers.
Thx!

**Comment by wooorm** [comment]
remark-math makes the processor *see* math, so remark-lint will ignore it. It doesn't convert, other plugins do that!

**Comment by Ir1d** [comment]
Thank you so much! I added "remark-math" to the plugins list, and remark-lint is perfect now. :+1:
